lib/ddtrace/configuration/components.rb

```diff
@@ -230,8 +231,11 @@ def build_tracer_tags(settings)
         end
 
         def build_test_mode_trace_flush(settings)
-          # If context flush behavior is provided, use it instead.
-          settings.test_mode.trace_flush || build_trace_flush(settings)
+          if settings.tracer.partial_flush.enabled
+            Datadog::CI::TraceFlush::Partial.new(min_spans_before_partial_flush: settings.tracer.partial_flush.min_spans_threshold)
+          else
+            Datadog::CI::TraceFlush::Finished.new
```

Comment:
Wait, is it me or this is a behavior change, on top of the cleanup?

If I'm following the previous version correctly, we used to use here the value from `settings.test_mode.trace_flush` which came from `lib/datadog/ci/configuration/components.rb` which always set this to `Datadog::CI::TraceFlush::Finished.new` (unless an override was manually provided).

Instead, we now introduce the `Partial` which I can't see being used before.

Obviously (and due to the change in the spec also) this looks like a deliberate change; could you add some detail on why this change was made as well? Was this just a bug in the previous version that got fixed?

Reply:
> Instead, we now introduce the Partial which I can't see being used before.

To use `Partial` before, you'd have to manually instantiate it at `c.test_mode.trace_flush = my_partial_instance`.

For the default tracer this was never the case: it instantiates either `Partial` or `Finished`  for you (based on the value of `c.tracer.partial_flush.enabled`).

This change won't affect clients using the default value, which is still `Datadog::CI::TraceFlush::Finished`.
Clients wanting to use partial flushing will use `c.tracer.partial_flush.enabled = true` instead of `c.test_mode.trace_flush = Datadog::CI::TraceFlush::Partial.new`.

Reply:
Thanks for the extra context. Seems reasonable!